dashboard/app/helpers/pd/workshop_survey_results_helper.rb

```diff
@@ -233,9 +233,10 @@ def generate_workshops_survey_summary(workshop)
               end
               session_summary[:facilitator][q_key] = facilitator_response_averages
             else
-              # For non facilitator specific responses, just return the average
-              sum = surveys_for_session.map {|survey| survey[response_section][q_key]}.reduce(0, :+)
-              session_summary[response_section][q_key] = (sum / surveys_for_session.size.to_f).round(2)
+              # For non facilitator specific responses, just return a histogram with nulls removed
+              summary = Hash[*surveys_for_session[response_section].map {|survey| survey[q_key]}.group_by {|v| v}.flat_map {|k, v| [k, v.size]}]
```

Comment:
This is a bit unwieldy. I'm having a hard time following what this is doing. Can we expand it out so it's more readable?

Reply:
AFAIK there isn't really a more elegant way to convert from array to frequency map, but I'll update the comment to make it a little more reasonable